### PR TITLE
Add definitions and autocompletions to js template function

### PR DESCRIPTION
**Pull request by arikmaor**

### CODE REVIEW

File: lib/functionTemplates/js/binaris.d.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+}
+
+declare class BinarisHttpResponse {
+  userResponse: Partial<FunctionResponse>;
+  constructor(userResponse: Partial<FunctionResponse>);
+}
+
+declare type FunctionContext = {
+  request: BinarisHttpRequest;
+  HTTPResponse: typeof BinarisHttpResponse;
```
Comment (michaeladda) [inline lib/functionTemplates/js/binaris.d.ts]: HTTP and Http. I don't have a preference, but I like consistency

File: lib/functionTemplates/js/binaris.d.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
+/// <reference types="node" />
+interface BinarisHttpRequest {
+  env: Record<string, string | undefined>;
+  query: Record<string, string | string[] | undefined>;
+  body: Buffer;
+  path: string;
+  method: string;
```
Comment (michaeladda) [inline lib/functionTemplates/js/binaris.d.ts]: this can be more restrictive ( 'GET' | 'PUT' | ...) but I'm not sure it's valuable

Comment (rylandg) [reply lib/functionTemplates/js/binaris.d.ts]: There are some non-standard headers which I assume we allow users to support via custom `HTTPResponse` such as `PATCH`.

File: lib/functionTemplates/js/binaris.d.ts [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  userResponse: Partial<FunctionResponse>;
+  constructor(userResponse: Partial<FunctionResponse>);
+}
+
+declare type FunctionContext = {
+  request: BinarisHTTPRequest;
+  HTTPResponse: typeof BinarisHTTPResponse;
+};
+
+export declare type BinarisFunction = (body: unknown, context: FunctionContext) => Promise<any>
```
Comment (rylandg) [inline lib/functionTemplates/js/binaris.d.ts]: Don't we know the generic form of the body?

Comment (arikmaor) [reply lib/functionTemplates/js/binaris.d.ts]: It can be whatever you pass in the body...

**Pull request merged**